### PR TITLE
[KR] '05-3.md' translated into Korean.

### PR DESCRIPTION
**Pull request by choongheelee**

### CODE REVIEW

**Comment by Atcold** [comment]
It looks good to me. You need to update the `_config.yml` file.
@skyil7, do you want to check the content?

File: docs/ko/week05/05-3.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+z_i &= 3y_i^2 = 3(x_i-2)^2 \\
+\frac{da}{dx_i} &= \frac{1}{4}\times3\times2(x_i-2) = \frac{3}{2}x_i-3 \\
+x &= \begin{pmatrix} 1&2\\3&4\end{pmatrix} \\
+\left(\frac{da}{dx_i}\right)^\top &= \begin{pmatrix} 1.5-3&3-3\\[2mm]4.5-3&6-3\end{pmatrix}=\begin{pmatrix} -1.5&0\\[2mm]1.5&3\end{pmatrix}
+\end{aligned}
+$$
+
+<!--Whenever you use partial derivative in PyTorch, you get the same shape of the original data. But the correct Jacobian thing should be the transpose.-->
+
+파이토치에서 편도함수를 이용할 때마다, 원래 데이터의 동일한 형태를 얻는다. 하지만 정확한 야코비안 방식은 다른 방식일 것이다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:175]: > 원래 데이터의 동일한 형태를 얻는다.
- 원래 데이터와 동일한 형태<sup>shape</sup>를 얻는다.

> 하지만 정확한 야코비안 방식은 다른 방식일 것이다.
> But the correct Jacobian thing should be the transpose.
- 하지만 정확한 야코비안<sup>Jacobian</sup> 형태는 이것의 전치행렬<sup>transpose</sup>이 되어야 한다.

File: docs/ko/week05/05-3.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+### 머리부터 발끝까지
+
+<!--Now we have a $1\times3$ vector $x$, assign $y$ to the double $x$ and keep doubling $y$ until its norm is smaller than $1000$. Due to the randomness we have for $x$, we cannot directly know the number of iterations when the procedure terminates.-->
+
+지금 우리에게 $1\times3$ 벡터 $x$가 있고, $y$를 $x$의 2배만큼 부여하는데 이때 $y$의 놈이 $1000$보다 작아질 때까지 계속 2배씩 부여한다. $x$에 대하여 우리가 가진 무작위성 때문에, 연산 과정이 끝날 때 반복 횟수를 바로 알 수 없다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:181]: > $y$를 $x$의 2배만큼 부여하는데 이때 $y$의 놈이 $1000$보다 작아질 때까지 계속 2배씩 부여한다.
> assign $y$ to the double $x$ and keep doubling $y$ until its norm is smaller than $1000$.
- $y$ 에 $x$ 를에 2를 곱한 값을 할당하고, $y$의 놈<sup>norm</sup>이 $1000$보다 작아질 때까지 계속 $y$의 값을 두배로 만드는 ($y$ 곱하기 2) 작업을 계속한다.

File: docs/ko/week05/05-3.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+y = x * 2
+i = 0
+while y.data.norm() < 1000:
+    y = y * 2
+    i += 1
+```
+
+<!--However, we can infer it easily by knowing the gradients we have.-->
+
+그러나, 우리가 가진 경사를 알고 있기 때문에 쉽게 추론할 수 있다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:195]: > 그러나, 우리가 가진 경사를 알고 있기 때문에 쉽게 추론할 수 있다.
- 그러나, 우리는 경사값을 알고 있기 때문에 이를 쉽게 추론할 수 있다.

File: docs/ko/week05/05-3.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+print(x.grad)
+tensor([1.0240e+02, 1.0240e+03, 1.0240e-01])
+print(i)
+9
+```
+
+<!--As for the inference, we can use `requires_grad=True` to label that we want to track the gradient accumulation as shown below. If we omit `requires_grad=True` in either $x$ or $w$'s declaration and call `backward()` on $z$, there will be runtime error due to we do not have gradient accumulation on $x$ or $w$.-->
+
+추론에 대하여 살펴보면, 아래에 코드에 나와있듯이 우리가 원하는 경사 축적을 따라가기 위한 자취를 남기기 위하여 `requires_grad=True`를 사용할 수 있다. 만약 $x$나 $w$ 선언에서 `requires_grad=True`를 빠뜨리고 $z$상의 `backward()`를 호출하면, $x$sk $w$상의 경사 축적이 없기 때문에 런타임 에러가 발생하게 된다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:209]: > 우리가 원하는 경사 축적을 따라가기 위한 자취를 남기기 위하여 `requires_grad=True`를 사용할 수 있다.
- `requires_grad=True` 사용하여 경사 축적을 추적해 나갈 수 있다.

Comment (marofmar) [reply docs/ko/week05/05-3.md:209]: > $x$sk $w$상의 경사 축적이 없기 때문에 런타임 에러가 발생하게 된다.
-  $x$ 또는 $w$ 에 축적된 경사가 없기 때문에 런타임 오류가 발생한다.

File: docs/ko/week05/05-3.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+
+```python
+# Both x and w that allows gradient accumulation
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:213]: 주석 번역 누락

File: docs/ko/week05/05-3.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+<!--And, we can have `with torch.no_grad()` to omit the gradient accumulation.-->
+
+그리고 이 경사 축적을 생략하고자 `with torch.no_grad()`를 쓸 수 있다.
+
+```python
+x = torch.arange(1., n + 1)
+w = torch.ones(n, requires_grad=True)
+
+# All torch tensors will not have gradient accumulation
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:229]: 주석 번역 누락

File: docs/ko/week05/05-3.md [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+with torch.no_grad():
+    z = w @ x
+
+try:
+    z.backward()  # PyTorch will throw an error here, since z has no grad accum.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:234]: 주석 번역 누락

File: docs/ko/week05/05-3.md [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+
+또한 기초적인 수치 연산 대신, 신경 그래프로 연결될 수 있는 자체 정의 모듈이나 함수들을 만들 수 있다. 주피터 노트북에서 작성한 코드는 [여기서]https://github.com/Atcold/pytorch-Deep-Learning/blob/master/extra/b-custom_grads.ipynb) 확인할 수 있다.
+
+<!--To do so, we need to inherit `torch.autograd.Function` and override `forward()` and `backward()` functions. For example, if we want to training nets, we need to get the forward pass and know the partial derivatives of the input respect to the output, such that we can use this module in any kind of point in the code. Then, by using back-propagation (chain rule), we can plug the thing anywhere in the chain of operations, as long as we know the partial derivatives of the input respect to the output.-->
+
+이를 위하여 `torch.autograd.Function`을 상속받아야 하고 `forward()`와 `backward()` 함수를 무시해야 한다. 예를 들어, 신경망을 학습시키고자 할 때, 순방향으로 진행되도록 해야하고 코드에서 어느 위치에서든 사용할 수 있는 모듈과 같은 출력값에 대한 입력 의 편도함수를 알아야 한다. 그리고 역전파(연쇄규칙)을 통해 출력값에 대한 편도함수를 알기 때문에 연속적인 연산이 일어나는 어느 지점에서든지 이를 연결할 수 있다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:249]: >  `forward()`와 `backward()` 함수를 무시해야 한다.
- `forward()`와 `backward()` 함수를 재정의<sup>override</sup> 해야 한다.

Comment (marofmar) [reply docs/ko/week05/05-3.md:249]: > 순방향으로 진행되도록 해야하고 코드에서 어느 위치에서든 사용할 수 있는 모듈과 같은 출력값에 대한 입력 의 편도함수를 알아야 한다.
- 순방향 진행 경로를 확보하고, 이 모듈을 코드의 어느 지점에서든 사용할 수 있도록 출력에 대한 입력의 편도함수를 알아야 한다.

Comment (marofmar) [reply docs/ko/week05/05-3.md:249]: > 그리고 역전파(연쇄규칙)을 통해 출력값에 대한 편도함수를 알기 때문에 연속적인 연산이 일어나는 어느 지점에서든지 이를 연결할 수 있다.
- 그 다음, 역전파<sup>back-propagation</sup>(연쇄 규칙)<sup>chain rule</sup>을 사용하여 출력에 대한 입력의 편도함수를 아는 어느 곳에나 이 모듈을 작업 체인에 연결시킬 수 있다.

File: docs/ko/week05/05-3.md [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+
+<!--In this case, there are three examples of ***custom modules*** in the *notebook*, the `add`, `split` and `max` modules. For example, the custom addition module:-->
+
+이 경우, *notebook* 안에 `add`, `split` 그리고 `max` 세 가지 맞춤 모듈이 있다. 맞춤형 추가 모듈은 예를 들면 아래와 같다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:253]: >  세 가지 맞춤 모듈이 있다.
- 세 가지 ***맞춤 모듈*** 이 있다.

File: docs/ko/week05/05-3.md [hunk 10]
```diff
@@ -0,0 +1,282 @@
...
+
+```python
+# Custom addition module
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:256]: 주석 번역 누락

File: docs/ko/week05/05-3.md [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+        grad_x1 = grad_output * torch.ones_like(x1)
+        grad_x2 = grad_output * torch.ones_like(x2)
+        # need to return grads in order
+        # of inputs to forward (excluding ctx)
+        return grad_x1, grad_x2
+```
+
+<!--If we have addition of two things and get an output, we need to overwrite the forward function like this. And when we go down to do back propagation, the gradients copied over both sides. So we overwrite the backward function by copying.-->
+
+우리가 두 가지 추가 사항을 가지고 출력 결과를 얻는다면, 이와 같은 순방향 함수에 덮어써야 한다. 그리고 역전파 단계로 내려갈 때, 경사는 양쪽 모두를 복사한다. 그래서 복사에 의한 역방향 함수를 덮어써야 한다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:278]: > 이와 같은 순방향 함수에 덮어써야 한다.
- 이런 방식으로 순방향 함수를 덮어써야 한다.

Comment (marofmar) [reply docs/ko/week05/05-3.md:278]: > 경사는 양쪽 모두를 복사한다. 그래서 복사에 의한 역방향 함수를 덮어써야 한다. 
- 경사는 양쪽 모두에 복사되고, 따라서 복사를 통해 역방향 함수를 덮어쓰게 된다.

File: docs/ko/week05/05-3.md [hunk 12]
```diff
@@ -0,0 +1,282 @@
...
+        return grad_x1, grad_x2
+```
+
+<!--If we have addition of two things and get an output, we need to overwrite the forward function like this. And when we go down to do back propagation, the gradients copied over both sides. So we overwrite the backward function by copying.-->
+
+우리가 두 가지 추가 사항을 가지고 출력 결과를 얻는다면, 이와 같은 순방향 함수에 덮어써야 한다. 그리고 역전파 단계로 내려갈 때, 경사는 양쪽 모두를 복사한다. 그래서 복사에 의한 역방향 함수를 덮어써야 한다. 
+
+<!--For `split` and `max`, see the code of how we overwrite forward and backward functions in the *notebook*. If we come from the same thing and **Split**, when go down doing gradients, we should add / sum them. For `argmax`, it selects the index of the highest thing, so the index of the highest should be $1$ while others being $0$. Remember, according to different custom modules, we need to overwrite its own forward pass and how they do gradients in backward function.-->
+
+`split`과 `max`에 대해서는 *notebook*에서 순방향, 역방향 함수를 덮어씌운 코드를 살펴볼 것을 권한다. **Split**과 동일한 것에서 출발한다면, 경사를 구하는 단계로 내려갈 때, 우리는 그것들을 추가하거나 더해야 한다. `argmax`의 경우, 가장 높은 것의 인덱스 값을 선택하므로, 가장 높은 것의 인덱스 값은 반드시 $1$이고 아닌 것은 $0$이어야 한다. 다른 맞춤 모듈에 따라 우리가 기억해야 할 것은 순방향으로 진행될 수 있도록 덮어씌워야 한다는 것과 역방향 함수에서 경사를 어떻게 하는지 여부이다.
```
Comment (marofmar) [inline docs/ko/week05/05-3.md:282]: > `split`과 `max`에 대해서는 *notebook*에서 순방향, 역방향 함수를 덮어씌운 코드를 살펴볼 것을 권한다.
- `split`과 `max`에 대해서는 *notebook*에서 우리가 어떻게 순방향, 역방향 함수를 덮어씌우는지에 대한 코드를 살펴볼 것을 권한다.

Comment (marofmar) [reply docs/ko/week05/05-3.md:282]: > 다른 맞춤 모듈에 따라 우리가 기억해야 할 것은 순방향으로 진행될 수 있도록 덮어씌워야 한다는 것과 역방향 함수에서 경사를 어떻게 하는지 여부이다.
> Remember, according to different custom modules, we need to overwrite its own forward pass and how they do gradients in backward function.
- 다양한 맞춤 모듈에서 우리는 이 모듈의 순방향 경로를 우리가 덮어써야 한다는 것과, 이들이 역방향 함수에서 어떻게 경사를 다루는지를 기억해야 한다.

**Comment by skyil7** [comment]
@Atcold Someone did!
@choongheelee Please update the content! and you should update `_config.yml` file in `docs/`.

That's the index file for website. It's not hard to handle. But call me if you need some help.

File: docs/ko/week05/05-3.md [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+출력 결과는 $n-k+1$ 사이즈에 $m$개의 층으로 이루어진 벡터이다.
+
+<center>
+<img src="{{site.baseurl}}/images/week05/05-3/Result_1D_Conv.png" alt="2" style="zoom:40%;" /><br>
+<b>그림 2</b>: 1D 합성곱 결과
+</center>
+
+<!--Furthermore, a single input vector can viewed as a monophonic signal.-->
+
+여기에 단일 입력 벡터는 모노 신호로 보일 수 있다.
```
[the remark this answers is not in the document]
Comment (choongheelee) [reply docs/ko/week05/05-3.md:38]: 네, 잘 알겠습니다. 예기치 않게 작업이 자꾸 늦어져 죄송합니다ㅠㅠ

**Comment by choongheelee** [comment]
요청하신 사항들을 반영하여 업데이트 완료하였습니다. 확인 부탁드리겠습니다.

Update complete with your requests. Please check this out if there's other issue.
